framework/src/org/checkerframework/framework/stub/StubParser.java

```diff
@@ -1387,148 +1425,173 @@ private AnnotationMirror getAnnotation(
         return annoMirror;
     }
 
-    /*
-     * Handles expressions in annotations.
-     * Supports String, int, and boolean literals, but not other literals
-     * as documented in the stub file limitation section of the manual.
-     */
-    private void handleExpr(AnnotationBuilder builder, String name, Expression expr) {
+    private Object getValueOfExpressionInAnnotation(
+            String name, Expression expr, TypeKind valueKind) {
         if (expr instanceof FieldAccessExpr || expr instanceof NameExpr) {
             VariableElement elem;
-            if (expr instanceof FieldAccessExpr) {
-                elem = findVariableElement((FieldAccessExpr) expr);
-            } else {
+            if (expr instanceof NameExpr) {
                 elem = findVariableElement((NameExpr) expr);
+            } else {
+                elem = findVariableElement((FieldAccessExpr) expr);
             }
-
             if (elem == null) {
-                // A warning was already issued by findVariableElement;
-                return;
+                // TODO: should this be stubAlwaysWarn, as it is an input stub file error?
```

Comment:
I added a few TODO comments for reporting input stub file errors properly. At the moment the user would get a rather scary looking error about a crash in the compiler. If the error is in the input stub file, stubAlwaysWarn seems more appropriate. But I didn't know what to return for that case and didn't want to restructure all this code.